### DISCUSSION TITLE
Static Analysis Techniques used by SpotBugs

### DISCUSSION THREAD

**Discussion by SeverinNovack** [spotbugs/spotbugs#2556] [Q&A]
Hello there, is there an overview of the static analysis techniques that are used in SpotBugs? Data Flow Analysis and Pattern Matching are definetly used but what about abstract interpretation and constraint based analysis?

**Comment by Pankraz76** [comment]


**Reply by JuditKnoll** [reply]
I don't see how this relates to the question. The question is about the techniques used in SpotBugs, not about other static analysis tools used by SpotBugs.

**Reply by Pankraz76** [reply]
ah right im sorry, yes.
